src/main/java/org/stellar/sdk/Sep10Challenge.java

```diff
@@ -40,4 +46,366 @@ public static String newChallenge(
+   * @param signers              The signers of client account.
+   * @return a list of signers that were found is returned, excluding the server account ID.
+   * @throws InvalidSep10ChallengeException If the SEP-0010 validation fails, the exception will be thrown.
+   * @throws IOException                    If read XDR string fails, the exception will be thrown.
+   */
+  public static List<String> verifyChallengeTransactionSigners(String challengeTransaction, String serverAccountId, Network network, List<String> signers) throws InvalidSep10ChallengeException, IOException {
+    if (signers == null || signers.isEmpty()) {
+      throw new InvalidSep10ChallengeException("No signers provided.");
+    }
+
+    // Read the transaction which validates its structure.
+    ChallengeTransaction readableChallengeTransaction = readChallengeTransaction(challengeTransaction, serverAccountId, network);
+    Transaction transaction = readableChallengeTransaction.getTransaction();
+
+    // Ensure the server account ID is an address and not a seed.
+    KeyPair serverKeyPair = KeyPair.fromAccountId(serverAccountId);
+
+    // Deduplicate the client signers and ensure the server is not included
+    // anywhere we check or output the list of signers.
+    List<String> clientSigners = new ArrayList<String>();
+    for (String signer : signers) {
+      // Ignore the server signer if it is in the signers list. It's
+      // important when verifying signers of a challenge transaction that we
+      // only verify and return client signers. If an account has the server
+      // as a signer the server should not play a part in the authentication
+      // of the client.
+      if (serverKeyPair.getAccountId().equals(signer) || clientSigners.contains(signer)) {
+        continue;
+      }
+      clientSigners.add(signer);
+    }
+
+    // Verify all the transaction's signers (server and client) in one
+    // hit. We do this in one hit here even though the server signature was
+    // checked in the readChallengeTx to ensure that every signature and signer
+    // are consumed only once on the transaction.
+    List<String> allSigners = new ArrayList<String>(clientSigners);
+    allSigners.add(serverKeyPair.getAccountId());
+    List<String> allSignersFound = verifyTransactionSignatures(transaction, allSigners);
+
+    // Confirm the server is in the list of signers found and remove it.
+    boolean serverSignerFound = false;
+    List<String> signersFound = new ArrayList<String>();
+    for (String signer : allSignersFound) {
+      if (serverKeyPair.getAccountId().equals(signer)) {
+        serverSignerFound = true;
+        continue;
+      }
+      signersFound.add(signer);
+    }
+
+    // Confirm we matched a signature to the server signer.
+    if (!serverSignerFound) {
+      throw new InvalidSep10ChallengeException(String.format("Transaction not signed by server: %s.", serverAccountId));
+    }
+
```

Comment:
I think we need to add a check here similar to the Go SDK that if the signers found after removing the SDK is length 0, then that should be an exception.  https://github.com/stellar/go/blob/a135d590c455964bf1bfdbb33ba266b5a08f9e53/txnbuild/transaction.go#L608-L610